src/core/environment.go

```diff
@@ -676,14 +676,32 @@ type Shell interface {
 	Exec(cmd string, arg ...string) ([]byte, error)
 }
 
-type execShellCommand struct{}
+type ExecShellCommand struct{}
 
-func (e execShellCommand) Exec(cmd string, arg ...string) ([]byte, error) {
+// FakeShell mocks shell command output and errors
+type FakeShell struct {
```

Comment:
It's a little unusual to include test mocks in the production code.

Looks like this project already has some packages dedicated to test code, how about we move `FakeShell` into https://github.com/nginx/agent/blob/main/test/utils/environment.go, and then use that package from our tests?

Reply:
Sure. I can move to utils package.